pages/1.11/security/ent/secrets/secrets-api/index.md

```diff
@@ -94,11 +97,42 @@ curl -H "Authorization: token=$(dcos config show core.dcos_acs_token)"
 Authentication tokens expire after five days by default. If your program needs to run longer than five days, you will need a service account. Please see [Provisioning custom services](/1.11/security/ent/service-auth/custom-service-auth/) for more information.
 
 
+## Text and binary secrets
+
+From DC/OS Enterprise version 1.11 the Secrets API supports text and binary secrets. A text secret is a text value encoded with UTF-8 encoding. The Secrets API passes through this value to a secret store backend without any modification. A binary secret is automatically encoded with base64 codec to a text format before it is passed to a secrets store backend.
+
+Use cases for secret types:
+
+* **Text**: can support storing text passwords, JSON documents, YAML documents
+
+* **Binary**: all the text secrets and also binary files, like Java KeyStore, Kerberos keytab files
+
+**Note**: It is not recommended to use binary secrets values as environment variables. The POSIX environment variables cannot contain NUL bytes.
+
+### Storing a secret
+
+When an HTTP client is sending a secret value to the Secrets API it should send `Content-Type` header containing information about a secret type. For backward compatibility with existing HTTP clients if a `PUT`/`PATCH` request is sent without the `Content-Type` header the Secrets API assumes a text secret and it behaves as if `application/json` was included in an request.
```

Comment:
"send `Content-Type` header" -> "send a `Content-Type` header"

Reply:
"assumes a text secret" - "assumes that the secret is a text secret", I think.

I'm not sure that we need to expose the reasoning (backward compatibility).